### ISSUE TITLE
Version 16.1 rc 0 failover not worked

### ISSUE THREAD

**Issue by gituser-4** [ethereum-mining/ethminer#1620]
**Desktop (please complete the following information):**
- Operating System:  Windows 10
- Hardware RX570 
- Ethminer Version 16.1 rc 0

**The problem:** when main pool is not worked, ethminer not switch to failover pool (failover work wery well if run as main pool). 

Logs of problem:

 ![error](https://user-images.githubusercontent.com/37613304/46516626-d8491180-c87b-11e8-8ab3-9ff024b27fde.JPG)

**Comment by AndreaLanfranchi** [comment]
@gituser-4 could you please report a log in english ? 
I cant' read the error messages

**Comment by gituser-4** [comment]
You can repeat this bug whis this bat file:


@echo off

title EtherDig PPS

cd /d "%~dp0"

:ethminer

timeout /t 20 && echo.

ethminer.exe --exit --farm-recheck 2000 -G -P stratum2+tcp://0x9ad1f252689cf813fdca9795b900bff265606b09.Rx570@etherdig.net:4444 -P stratum+ssl://0x9ad1f252689cf813fdca9795b900bff265606b09.Fail570:x@eu1.ethermine.org:5555 --tstop 80 --tstart 45 --api-port 3333 --failover-timeout 15 --cl-local-work 256 --cl-global-work 65536   


echo Restart at %date% %time:~0,8% >> log_%date%.txt

goto ethminer

**Comment by AndreaLanfranchi** [comment]
@chfast and @gituser-4 
There is **no bug** going on here as ethminer is working correctly according to the CLI directives entered.

Reading the log it fails and tries to reconnect to `etherdig.net` for **3 times** before selecting failover `eu1.ethermine.org`. And this is what the default value (3) for directive `--farm-retries` dictates: ethminer tries to reconnect to same pool for the number of times specified in `--farm-retries`.

If you want have a quicker switch-over please specify a lower value for `--farm-retries`.
(always read `ethminer --help` fully) 

One last note : **do not set `--farm-recheck` as in stratum mode it does nothing.**

**Comment by gituser-4** [comment]
And I can not see 3 times connect to failover eu1.ethermine.org!

**Comment by XhmikosR** [comment]
@gituser-4: please be patient, @AndreaLanfranchi will answer your question.

**Comment by gituser-4** [comment]
I am specify --farm-retries 1 and BUG IS REPEATED! 
Ethminer try 1 times connect to etherdig.net and not connect to failover! 
Are you try my bat-file? 
Are you read what I write?

**Comment by smurfy** [comment]
ethmine is your failover and etherdig is your main.

it tries to etherdig and then tries ethermine. But at least in your log you exit ethminer before it tries to connect. It only says selected.

Also since the last message after the ethermine message is in cyrillic we cant read what it says.

**Comment by AndreaLanfranchi** [comment]
You're too impatient. Yes I tried your parameters and they **DO WORK AS EXPECTED**.

Follow this. Ethminer starts and picks the first pool in list (the PRIMARY)
```
Selected pool etherdig.net:4444
```
at this stage it tries to connect but it gets error and shows
```
Disconnected from [190.115.26.22:4444]
```
It repeats this cycle 3 times (as specified by default value of --farm-retries).

After that it shows
```
Selected pool eu1.ethermine.org:5555
```
here it begins to connect to the failover but you have stopped the log so you do not see any connection.

**Comment by gituser-4** [comment]
In cyrilluc output for command:

timeout /t 20 

Ethminer write "selected pool eu1.ethermine.org:5555" and EXIT! 
NOT CONNECT!

**Comment by gituser-4** [comment]
"here it begins to connect to the failover but you have stopped the log"

I am NOT stopped log! 
Ethminer EXIT! And log stopped too.

**Comment by gituser-4** [comment]
In cyrilluc output for command:

timeout /t 20 

Ethminer write "selected pool eu1.ethermine.org:5555" and EXIT! 
NOT try to connect!

**Comment by gituser-4** [comment]
"here it begins to connect to the failover but you have stopped the log"

I not stopped log! You do not understand.
Ethminer EXIT! And log stopped too

**Comment by smurfy** [comment]
````
m 13:02:40 main     ethminer 0.16.1
 m 13:02:40 main     Build: windows/release
 i 13:02:40 main     Found suitable OpenCL device [GeForce GTX 1080] with 8,000 GB of GPU memory
 i 13:02:40 main     Configured pool etherdig.net:4444
 i 13:02:40 main     Configured pool eu1.ethermine.org:5555
 i 13:02:40 main     Api server listening on port 3333.
 i 13:02:40 main     Selected pool etherdig.net:4444
 X 13:02:42 stratum  Error  190.115.26.22:4444 [ No connection could be made because the target machine actively refused it ]
 X 13:02:42 stratum  No more IP addresses to try for host: etherdig.net
 i 13:02:42 main     Disconnected from  [190.115.26.22:4444]
 i 13:02:42 main     Selected pool etherdig.net:4444
 X 13:02:43 stratum  Error  190.115.26.22:4444 [ No connection could be made because the target machine actively refused it ]
 X 13:02:43 stratum  No more IP addresses to try for host: etherdig.net
 i 13:02:43 main     Disconnected from  [190.115.26.22:4444]
 i 13:02:44 main     Selected pool etherdig.net:4444
 m 13:02:45 main     not-connected
 X 13:02:45 stratum  Error  190.115.26.22:4444 [ No connection could be made because the target machine actively refused it ]
 X 13:02:45 stratum  No more IP addresses to try for host: etherdig.net
 i 13:02:45 main     Disconnected from  [190.115.26.22:4444]
 i 13:02:46 main     Selected pool eu1.ethermine.org:5555
````
i can reproduce the problem on windows.

**Comment by smurfy** [comment]
Compiling ``release/0.16`` branch right now. to see whats up

**Comment by gituser-4** [comment]
Oh my god! 
It happened!

**Comment by smurfy** [comment]
Its somewhat related to ssl.  If i change port and to non ssl  it works as expected.
So this could be a workaround for @gituser-4

**Comment by gituser-4** [comment]
Ok, thank you!

But if I use this failover pool whith ssl as main pool, it work wery well.

**Comment by smurfy** [comment]
@XhmikosR yes, the problem exists in master too. I created a PR with a possible fix for the problem.
